axelrod/fingerprint.py

```diff
@@ -14,219 +16,355 @@
-        Points with coordinates that will be closer together.
-    progress_bar : bool
-        Whether or not to create a progress bar which will be updated
+    def plot(self, col_map='seismic', interpolation='none', title=None,
+             colorbar=True, labels=True):
+        """Plot the results of the spatial tournament.
 
-    Returns
-    ----------
-    points : list
-        of Point objects with coordinates (x, y)
-    """
-    num = int((1 / step) // 1) + 1
+        Parameters
+        ----------
+        col_map : str, optional
+            A matplotlib colour map, full list can be found at
+            http://matplotlib.org/examples/color/colormaps_reference.html
+        interpolation : str, optional
+            A matplotlib interpolation, full list can be found at
+            http://matplotlib.org/examples/images_contours_and_fields/interpolation_methods.html
+        title : str, optional
+            A title for the plot
+        colorbar : bool, optional
+            Choose whether the colorbar should be included or not
+        labels : bool, optional
+            Choose whether the axis labels and ticks should be included
 
-    if progress_bar:
-        p_bar = tqdm.tqdm(total=num ** 2, desc="Generating points")
+        Returns
+        ----------
+        figure : matplotlib figure
+            A heat plot of the results of the spatial tournament
+        """
 
-    points = []
-    for x in np.linspace(0, 1, num):
-        for y in np.linspace(0, 1, num):
-            points.append(Point(x, y))
+        plotting_data = self._data.get_plotting_data()
+        fig, ax = plt.subplots()
+        cax = ax.imshow(
+            plotting_data, cmap=col_map, interpolation=interpolation)
 
-            if progress_bar:
-                p_bar.update()
+        if colorbar:
+            max_score = max(self.data.values())
+            min_score = min(self.data.values())
+            ticks = [min_score, (max_score + min_score) / 2, max_score]
+            fig.colorbar(cax, ticks=ticks)
 
-    if progress_bar:
-        p_bar.close()
+        plt.xlabel('$x$')
+        plt.ylabel('$y$', rotation=0)
+        ax.tick_params(axis='both', which='both', length=0)
+        plt.xticks([0, len(plotting_data) - 1], ['0', '1'])
+        plt.yticks([0, len(plotting_data) - 1], ['1', '0'])
 
-    return points
+        if not labels:
+            plt.axis('off')
 
+        if title is not None:
+            plt.title(title)
+        return fig
 
-class AshlockFingerprint():
-    def __init__(self, strategy, probe=axl.TitForTat):
+
+def _get_play_kwargs(filename: Any, in_memory: bool,
+                     processes: Union[int, None], progress_bar: bool
+                     ) -> Dict[str, Any]:
+    """Adjust and build the kwargs for tournament.play() based on issues
+    with Windows OS."""
+    play_kwargs = {'build_results': False,
+                   'processes': processes,
+                   'progress_bar': progress_bar}
+
```

Comment:
how is this different from what Tournament already does?  How does Tournament usually handle `in_memory=None, file_name=None` ?